runelite-client/src/main/java/net/runelite/client/plugins/devtools/WidgetSearch.java

```diff
@@ -0,0 +1,245 @@
+					searchTerms.put("Type", Integer.parseInt(segments[i + 1]));
+					break;
+				case "contenttype":
+					searchTerms.put("ContentType", Integer.parseInt(segments[i + 1]));
+					break;
+				case "parentid":
+					searchTerms.put("ParentId", Integer.parseInt(segments[i + 1]));
+					break;
+				case "selfhidden":
+					if (segments[i + 1].equals("true"))
+					{
+						searchTerms.put("SelfHidden", true);
+					}
+					else
+					{
+						searchTerms.put("SelfHidden", false);
+					}
+					break;
+				case "hidden":
+					if (segments[i + 1].equals("true"))
+					{
+						searchTerms.put("Hidden", true);
+					}
+					else
+					{
+						searchTerms.put("Hidden", false);
+					}
+					break;
+				case "text":
+					searchTerms.put("Text", segments[i + 1]);
+					break;
+				case "textcolor":
+					searchTerms.put("TextColor", segments[i + 1]);
+					break;
+				case "name":
+					searchTerms.put("Name", segments[i + 1]);
+					break;
+				case "itemid":
+					searchTerms.put("ItemId", Integer.parseInt(segments[i + 1]));
+					break;
+				case "itemquantity":
+					searchTerms.put("ItemQuantity", Integer.parseInt(segments[i + 1]));
+					break;
+				case "modelid":
+					searchTerms.put("ModelId", Integer.parseInt(segments[i + 1]));
+					break;
+				case "spriteid":
+					searchTerms.put("SpriteId", Integer.parseInt(segments[i + 1]));
+					break;
+				case "width":
+					searchTerms.put("Width", Integer.parseInt(segments[i + 1]));
+					break;
+				case "height":
+					searchTerms.put("Height", Integer.parseInt(segments[i + 1]));
+					break;
+				case "relativex":
+					searchTerms.put("RelativeX", Integer.parseInt(segments[i + 1]));
+					break;
+				case "relativey":
+					searchTerms.put("RelativeY", Integer.parseInt(segments[i + 1]));
+					break;
+				case "canvaslocation":
+					String canvasSegments[] = segments[i + 1].split("[,]");
+					searchTerms.put("CanvasLocation", new Point(Integer.parseInt(canvasSegments[0]), Integer.parseInt(canvasSegments[1])));
+					break;
+				case "bounds":
+					String boundsSegments[] = segments[i + 1].split("[,]");
+					searchTerms.put("Bounds", new Rectangle(Integer.parseInt(boundsSegments[0]), Integer.parseInt(boundsSegments[1]), Integer.parseInt(boundsSegments[2]), Integer.parseInt(boundsSegments[3])));
+					break;
+				case "scrollx":
+					searchTerms.put("ScrollX", Integer.parseInt(segments[i + 1]));
+					break;
+				case "scrolly":
+					searchTerms.put("ScrollY", Integer.parseInt(segments[i + 1]));
+					break;
+				case "originalx":
+					searchTerms.put("OriginalX", Integer.parseInt(segments[i + 1]));
+					break;
+				case "originaly":
+					searchTerms.put("OriginalY", Integer.parseInt(segments[i + 1]));
+					break;
+				case "paddingx":
+					searchTerms.put("PaddingX", Integer.parseInt(segments[i + 1]));
+					break;
+				case "paddingy":
+					searchTerms.put("PaddingY", Integer.parseInt(segments[i + 1]));
+					break;
+			}
+		}
+	}
+
+	public boolean isMatch(Widget widget)
```

Comment:
I would change this around a bit, because you are expecting all search terms to match. As soon as something doesn't match you can already return a value, in this case `false`.